### PR TITLE
multi line interactive on {} working (a bit ugly)

### LINKED ISSUE #13: multiline support in stdin repl
right now when not passing in script files, so in repl mode, each line must stand by itself, that's bad

### PR DESCRIPTION
**Pull request by ldemailly**
fixes #13

### CODE REVIEW

File: lexer/lexer.go [hunk 1]
```diff
@@ -7,15 +7,22 @@ import (
...
+	input    string
+	pos      int
+	lineMode bool
 }
 
+// Mode with input expected the be complete (multiline/file).
 func New(input string) *Lexer {
 	l := &Lexer{input: input}
 	return l
 }
```
Comment (ccoVeille) [inline lexer/lexer.go:19]:
```suggestion
 func New(input string) *Lexer {
 	return &Lexer{input: input}
 }
```

**Review by ccoVeille — COMMENTED** [review]
LGTM 👍

Minor stylistic suggestions anyway

File: lexer/lexer.go [hunk 2]
```diff
@@ -86,7 +93,11 @@ func (l *Lexer) NextToken() token.Token { //nolint:funlen // many cases to lex.
 	case '"':
 		return token.Token{Type: token.STRING, Literal: l.readString()}
 	case 0:
-		return token.Token{Type: token.EOF, Literal: ""}
+		if l.lineMode {
+			return token.Token{Type: token.EOL}
+		} else {
+			return token.Token{Type: token.EOF}
+		}
```
[the remark this answers is not in the document]
Comment (ldemailly) [reply lexer/lexer.go:100]: I think we went through that last time, when there is exactly 1 line and it looks symetrical, while I'm very anti else in general, I don't in this case :)

File: lexer/lexer.go (hunk 1 above)
Comment (ldemailly) [reply lexer/lexer.go:19]: fixed in https://github.com/grol-io/grol/commit/5b28ec0cef422e46b5ba9867ffb7726e18b4c192
was meant to be an PR but I need to fix the permissions

File: lexer/lexer.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (ccoVeille) [reply lexer/lexer.go:100]: We might have talked about indeed.

But let's say you didn't convince me, and I didn't convince you either 😅🤭🤣

File: lexer/lexer.go (hunk 1 above)
Comment (ccoVeille) [reply lexer/lexer.go:19]: > not sure it's how it is, must have had something else 

As far as I know, only simple types like int, string need a variable.

https://github.com/golang/go/issues/45624

**Pull request merged**